com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/LuaBuilder.java

```diff
@@ -333,14 +332,21 @@ public void build(Task<Void> task) throws CompileExceptionError, IOException {
         LuaSource.Builder srcBuilder = LuaSource.newBuilder();
         srcBuilder.setFilename(task.input(0).getPath());
 
-        if (use_lua_source) {
+        // for platforms using Lua 5.1 we include Lua source code even though
+        // there is a constructLuaBytecode() function above
+        // tests have shown that Lua 5.1 bytecode becomes larger than source,
+        // even when compressed using lz4
+        // this is unacceptable for html5 games where size is a key factor
+        // see https://github.com/defold/defold/issues/6891 for more info
+        if (platformUsesLua51.contains(project.getPlatform())) {
             srcBuilder.setScript(ByteString.copyFrom(script.getBytes()));
-        } else if (needsVanillaLua32.contains(project.getPlatform())) {
-            byte[] bytecode = constructLuaBytecode(task, "luac-32", script);
-            if (bytecode != null) {
-                srcBuilder.setBytecode(ByteString.copyFrom(bytecode));
-            }
-        } else {
+        }
+        // include uncompressed Lua source code instead of bytecode
+        // see https://forum.defold.com/t/urgent-need-help-i-have-huge-problem-with-game-submission-to-apple/68031
+        else if (useUncompressedLuaSource) {
```

Comment:
isn't it just one more option for previous variant? you made these too IFs instead of `||` to split these cases?

Reply:
Correct. It is two different scenarios resulting in the same outcome. I wanted to separate them to make it clearer.